src/toil/test/src/jobCacheTest.py

```diff
@@ -0,0 +1,848 @@
+# Copyright (C) 2015 UCSC Computational Genomics Lab
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Author : Arjun Arkal Rao
+Affiliation : UCSC BME, UCSC Genomics Institute
+File : src/toil/test/src/JobCacheTest.py
+"""
+from __future__ import print_function
+import collections
+import os
+import sys
+import random
+import signal
+import time
+import unittest
+
+from struct import pack, unpack
+from uuid import uuid4
+
+from toil.job import Job, CacheError
+from toil.test import ToilTest, needs_aws, needs_azure
+from toil.leader import FailedJobsException
+from toil.jobStores.abstractJobStore import NoSuchFileException
+
+# Some tests take too long on the AWS and Azure Job stores and are unquitable for CI.  They can be
+# be run during manual tests by setting this to False.
+testingIsAutomatic = True
+
+
+class Hidden:
+    """
+    Hiding the abstract test class from the Unittest loader so it can be inherited in different test
```

Comment:
Hiding doesn't make inheritance possible, it simply prevents unittest from trying to instantiate it.